File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int countElement(vector<int>& nums) {
        int count = 0;
        int candidate = 0;

        for (auto num : nums) {
            if (count == 0) {
                candidate = num;
            }

            count += (num == candidate) ? 1 : -1;
        }
        return candidate;
    }
    int majorityElement(vector<int>& nums) { return countElement(nums); }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;

        int count = 1;

        int last = nums[0];

        for (int i = 1; i < n; i++) {
            if (count == 0) {
                last = nums[i];
            }

            
            if (nums[i] == last)
                count++;
            else {
                count--;
            }
        }
        return last;

    }
}